### PR TITLE
Ros2 and 2024 wpilib updates

### PR DESCRIPTION
**Pull request by armadilloz**

### CODE REVIEW

File: src/main/java/org/frc/team4048/rosnetworktables/NtRosProxy.java [hunk 1]
```diff
@@ -2,105 +2,74 @@
...
-                    System.out.println("WAITING");
-                    st=System.currentTimeMillis();
-               }
-               sleep(100);
-          }
-          initNetworkTables();
-          initializeTopics();
-          topics.start();
-     }
+    //TODO Add timeout
```
Comment (bengold22) [inline src/main/java/org/frc/team4048/rosnetworktables/NtRosProxy.java:35]: While you're in this code, is this comment legitimate? What would you be timing out?

**Comment by bengold22** [comment]
Only other observation I see is that it looks like there's various conditions that will throw, including Launcher (not changed in this PR). It might be useful to catch the error in Launcher, write it to system err, then exit. If something is getting output when it fails that's probably good enough.

There's also the comment about not being a stop(), I assume this is okay as it's designed to just stop on termination of the program.

File: src/main/java/org/frc/team4048/rosnetworktables/NtToRTopic.java [hunk 2]
```diff
@@ -26,27 +31,34 @@ protected NtToRTopic(NetworkTableInstance ntInst, Subscriber ntSubscriber,Publis
     public void start() {
         // Register the NT subscriber to messages
         ntInst.addListener(ntSubscriber, EnumSet.of(NetworkTableEvent.Kind.kValueAll),
-// TODO this is bad practice needs to be look at
+                // TODO this is bad practice needs to be look at
```
Comment (bengold22) [inline src/main/java/org/frc/team4048/rosnetworktables/NtToRTopic.java:34]: What is "bad practice" here?

**Review by bengold22 — APPROVED** [review]
I left various comments but nothing seems critical.

File: src/main/java/org/frc/team4048/rosnetworktables/NtRosProxy.java (hunk 1 above)
Comment (armadilloz) [reply src/main/java/org/frc/team4048/rosnetworktables/NtRosProxy.java:35]: Done.

File: src/main/java/org/frc/team4048/rosnetworktables/NtToRTopic.java (hunk 2 above)
Comment (armadilloz) [reply src/main/java/org/frc/team4048/rosnetworktables/NtToRTopic.java:34]: The cast to (T). It's because the way the NT API is built.

**Comment by armadilloz** [comment]
Added some exception handling at the top level (that's only called in case of exception, since the main thread terminates immediately and the other threads remain running)

**Pull request merged**